docs/developer/tutorials/logs/submit-logs-from-http.md

```diff
@@ -0,0 +1,142 @@
+```
+
+Then run:
+```
+ddev env start acme py3.11 --base
+```
+
+## Define an Agent Check
+
+We start by registering an implementation for our integration.
+At first it is empty, we will expand it step by step.
+
+Open `datadog_checks/acme/check.py` in our editor and put the following there:
+
+```python
+from datadog_checks.base.checks.logs.crawler.base import LogCrawlerCheck
+
+
+class AcmeCheck(LogCrawlerCheck):
+    __NAMESPACE__ = 'acme'
+```
+
+Now we'll run something we will refer to as *the check command*:
+```
+ddev env agent acme py3.11 check
+```
+
+We'll see the following error:
+```
+Can't instantiate abstract class AcmeCheck with abstract method get_log_streams
+```
+
+We need to define the `get_log_streams` method.
+As [its docs say](../../base/logs-crawlers.md#datadog_checks.base.checks.logs.crawler.base.LogCrawlerCheck.get_log_streams) it must return an iterator over `LogStream` subclasses.
+That's what the next section is about.
+
+## Define a Stream of Logs
+
+In the same file let's add a `LogStream` subclass and return it (wrapped in a list) from `AcmeCheck.get_log_streams`:
+
+```python
+from datadog_checks.base.checks.logs.crawler.base import LogCrawlerCheck
+from datadog_checks.base.checks.logs.crawler.stream import LogStream
+
+class AcmeCheck(LogCrawlerCheck):
+    __NAMESPACE__ = 'acme'
+
+    def get_log_streams(self):
+        return [AcmeLogStream(check=self, name='ACME log stream')]
+
+class AcmeLogStream(LogStream):
+    """Stream of Logs from ACME"""
+```
+
+Every time we modify the source code, make sure to restart the environment:
+
+```
+ddev env stop acme py3.11 && ddev env start acme py3.11 --base
+```
+
+Now running *the check command* will show a new error:
+
+```
+TypeError: Can't instantiate abstract class AcmeLogStream with abstract method records
+```
+
+Once again we need to define a method, this time [`LogStream.records`](../../base/logs-crawlers.md#datadog_checks.base.checks.logs.crawler.stream.LogStream.records).
+
+
+```python
+from datadog_checks.base.checks.logs.crawler.stream import LogRecord, LogStream
+
+... # Skip AcmeCheck to focus on LogStream.
+
+
+class AcmeLogStream(LogStream):
+    """Stream of Logs from ACME"""
+
+    def records(self):
+        return [
+            LogRecord(
+                data={'message': 'This is a log from ACME.', 'level': 'info'},
+                cursor={'timestamp': dt.now().timestamp()},
+            )
+        ]
+```
+
+There are several things going on here.
+`AcmeLogStream.records` returns an iterator over `LogRecord` objects.
+We discuss these objects first so to keep the method itself simple we return a list with just one record.
+
+### What is a Log Record?
+
+The `LogRecord` class has 2 fields.
+In `data` we put any data in here that we want to submit as a log to Datadog.
+In `cursor` we store a unique identifier for this specific `LogRecord`.
+Some things to consider when defining cursors:
+
+- Use UTC time stamps!
```

Comment:
nit:
Do we support all version of the epoch timestamp? Seconds, milli, nano etc?

Reply:
Yup, entirely up to the user what level of precision they choose!